raiden-ts/src/raiden.ts

```diff
@@ -1129,6 +1129,54 @@ export class Raiden {
     return receipt.transactionHash as Hash;
   }
 
+  /**
+   * Registers and creates a new token network for the provided token address.
+   * Throws an error, if
+   * <ol>
+   *  <li>Executed on main net</li>
+   *  <li>`token` is not a valid address</li>
+   *  <li>Token is already registered</li>
+   *  <li>Token could not be registered</li>
+   * </ol>
+   *
+   * @param token - Address of the token to be registered
+   * @param channelParticipantDepositLimit - The deposit limit per channel participant
+   * @param tokenNetworkDepositLimit - The deposit limit of the whole token network
+   * @returns Address of new token network
+   */
+  public async registerToken(
+    token: string,
+    channelParticipantDepositLimit: BigNumberish = MaxUint256,
+    tokenNetworkDepositLimit: BigNumberish = MaxUint256,
+  ): Promise<Address> {
+    // Check whether address is valid
+    assert(Address.is(token), [ErrorCodes.DTA_INVALID_ADDRESS, { token }], this.log.info);
+
+    // Check whether we are on a test network
+    assert(this.deps.network.chainId !== 1, ErrorCodes.RDN_REGISTER_TOKEN_MAINNET, this.log.info);
+
+    const { signer } = chooseOnchainAccount(this.deps, this.config.subkey);
+
+    const registryAddress = this.deps.contractsInfo.TokenNetworkRegistry.address;
+    const tokenNetworkRegistry = TokenNetworkRegistry__factory.connect(registryAddress, signer);
+
+    // Check whether token is already registered
+    const isRegistered = token in this.state.tokens;
+    assert(!isRegistered, ErrorCodes.RDN_REGISTER_TOKEN_REGISTERED, this.log.info);
```

Comment:
```suggestion
    await this.monitorToken(token).then((tokenNetwork) => {
      throw new RaidenError(ErrorCodes.RDN_REGISTER_TOKEN_REGISTERED, { tokenNetwork });
    }, () => undefined);
```
This way, you ensure we use `registryContract.token_to_token_networks`, since we don't necessarily keep all tokens in `state.tokens`, auto-dispatch `tokenMonitored` in case it's already there, and get a better `details` in error with the tokenNetwork address;